runtime_scan/pkg/orchestrator/scanresultwatcher/watcher.go

```diff
@@ -396,3 +399,50 @@ func (w *Watcher) cleanupResources(ctx context.Context, scanResult *models.Targe
 
 	return nil
 }
+
+// nolint:cyclop
+func (w *Watcher) reconcileAborted(ctx context.Context, scanResult *models.TargetScanResult) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+
+	scanResultID, ok := scanResult.GetID()
+	if !ok {
+		return errors.New("invalid ScanResult: ID is nil")
+	}
+
+	// Check if ScanResult is in aborted state for more time than the timeout allows
+	if scanResult.Status == nil || scanResult.Status.General == nil {
+		return errors.New("invalid ScanResult: Status or General is nil")
+	}
+
+	var transitionTimeToAbort time.Time
+	if scanResult.Status.General.LastTransitionTime != nil {
```

Comment:
if scanResult.Status.General.LastTransitionTime == nil should we return error? or just not continue?

Reply:
Nope. We definitely want to carry on in both cases to avoid having `ScanResults` stuck in `aborted` state.